src/qibo/models/variational.py

```diff
@@ -82,14 +83,12 @@ def _loss(params, circuit, hamiltonian):
 
         if method == 'sgd':
             # check if gates are using the MatmulEinsum backend
-            from qibo import K
-            if K.name == "custom":
+            if K.op is not None:
```

Comment:
If I understand, K.op is None for tensorflow and numpy, right? Does this work for numpy?

Reply:
Indeed `K.op` is `None` for all backends without custom operators (currently tensorflow and numpy). On the other hand SGD should only work with the `tensorflow` backend. I decided to remove this check completely because if you see in the `sgd` method defined in `optimizers.py` we already check if an SGD-compatible backend is used. I believe it is cleaner to have the check inside the sgd method instead of every model we call sgd.